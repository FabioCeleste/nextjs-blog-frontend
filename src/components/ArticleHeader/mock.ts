import { Author } from '../../shared-typed/author';
import { Category } from '../../shared-typed/category';

export default {
  id: '1',
  blogName: "Fabio's Blog",
  blogDescription: 'O Blog do Fabinho.',
  logo: '/assets/images/logo.png',
  showText: true,
  content:
    '<p style="margin-left:0px;text-align:justify;">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque rhoncus maximus ex, id auctor lacus dignissim a. Nulla eu leo placerat, varius dui id, euismod ex. Integer tempus nulla dui, id gravida sapien feugiat et. Mauris molestie elementum lacus sit amet aliquet. Nam mollis convallis massa, sit amet commodo nisi facilisis in. Sed rhoncus risus tellus, sollicitudin luctus mauris molestie id. Donec venenatis sodales orci, ut aliquet ex laoreet quis. Sed quis mauris a augue pulvinar imperdiet. Etiam feugiat libero purus, id lobortis libero lacinia vitae. Duis hendrerit augue magna, et pellentesque ipsum interdum sit amet.</p><p style="margin-left:0px;text-align:justify;">&nbsp;</p><p style="margin-left:0px;text-align:justify;">Praesent auctor nisi a laoreet placerat. Integer nibh turpis, egestas vel orci ac, sodales ultrices velit. Morbi justo urna, blandit vitae mattis at, sollicitudin ac dui. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. In hac habitasse platea dictumst. Quisque ac nisl felis. Morbi aliquam erat et turpis maximus, non ultrices nulla aliquet. Donec porttitor tortor non nisl feugiat condimentum. Sed fringilla pharetra tellus vel semper. Aliquam ut arcu odio. Etiam porttitor purus vitae tellus scelerisque, sit amet pellentesque ligula molestie.</p><p style="margin-left:0px;text-align:justify;">Ut faucibus odio diam, nec egestas tortor tincidunt non. Nulla volutpat enim et dolor scelerisque, sed vehicula justo fermentum. Cras ut condimentum ex. Fusce ac enim convallis, ultrices nisl sit amet, porttitor urna. Etiam ultrices ipsum a pharetra gravida. Integer suscipit sem et interdum consequat. Aenean commodo pretium dui sit amet fringilla. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Maecenas cursus odio id nisi aliquam vehicula. Morbi et nunc tellus. Sed vel turpis tincidunt, finibus magna ac, venenatis ante. Nulla in nulla ipsum. Curabitur efficitur tellus vestibulum arcu feugiat, vel iaculis nisi venenatis.</p><p style="margin-left:0px;text-align:justify;">&nbsp;</p><p style="margin-left:0px;text-align:justify;">Aliquam vel diam purus. Nam auctor dolor id magna cursus, a ornare felis commodo. Integer in auctor nibh. Mauris pharetra bibendum aliquet. Phasellus at justo dui. Sed ligula orci, varius eget sem vitae, porttitor aliquam mi. Sed justo ipsum, interdum non tempor vitae, viverra ac eros. Ut malesuada, nisi in aliquam interdum, diam diam vulputate lorem, et lacinia turpis mauris id odio. Donec sem arcu, convallis at blandit vitae, faucibus vitae magna. Aliquam scelerisque elit urna, a cursus justo scelerisque a. Quisque lacinia mauris eget venenatis euismod. Donec ac est quis nulla commodo rhoncus at sit amet arcu. Nulla viverra, lacus eget sodales eleifend, justo augue tempus leo, id ornare turpis orci vel metus. Integer turpis quam, ultrices ut vulputate nec, viverra sit amet risus. Suspendisse rutrum est in tincidunt porttitor.</p>',

  title: 'Primeiro Post',
  excerpt:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque rhoncus maximus ex, id auctor lacus dignissim a. Nulla eu',

  cover: {
    data: {
      id: '4',
      attributes: {
        url: '/assets/images/logo.png',
        alternativeText: 'logo.png',
      },
    },
  },

  createdAt: '2022-01-28T02:53:53.880Z',
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'Fabio Celeste',
        slug: 'fabio-celeste',
      },
    },
  } as Author,
  category: {
    data: {
      id: '1',
      attributes: {
        displayName: 'JavaScript',
        slug: 'java-script',
      },
    },
  } as Category,
};
