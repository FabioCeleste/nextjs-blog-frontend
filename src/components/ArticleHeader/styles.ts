import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;

    .post-cover {
      display: flex;
      flex-direction: column;
      align-items: center;
    }

    img {
      max-width: 100%;
      height: 480px;
    }

    h1 {
      font-size: ${theme.font.sizes.huge};
      color: ${theme.colors.darkText};
    }

    h2 {
      font-size: ${theme.font.sizes.medium};
      font-weight: normal;
      color: ${theme.colors.darkerGray};
    }

    .separator-header {
      width: 80%;
      height: 1px;
      background-color: ${theme.colors.mediumGray};
    }
  `}
`;
