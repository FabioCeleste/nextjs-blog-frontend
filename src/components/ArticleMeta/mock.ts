import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-01-28T02:53:53.880Z',
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'Fabio Celeste',
        slug: 'fabio-celeste',
      },
    },
  },
  category: {
    data: {
      id: '1',
      attributes: {
        displayName: 'JavaScript',
        slug: 'java-script',
      },
    },
  },
} as ArticleMetaProps;
