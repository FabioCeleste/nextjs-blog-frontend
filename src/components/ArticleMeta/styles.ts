import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: 16px;
    color: ${theme.colors.darkerGray};
    font-style: italic;

    a {
      color: ${theme.colors.secondaryColor};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    &:hover {
      filter: brightness(50%);
    }
  `}
`;
