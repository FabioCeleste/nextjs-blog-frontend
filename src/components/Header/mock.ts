import { HeaderProps } from '.';

export default {
  blogName: "Fabio's Blog",
  blogDescription: 'O Blog do Fabinho.',
  logo: '/assets/images/logo.png',
  showText: true,
} as HeaderProps;
