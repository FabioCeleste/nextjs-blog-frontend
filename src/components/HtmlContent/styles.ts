import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: 35px;

    p {
      font-size: 16px;
      font-weight: 600;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondaryColor};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    pre {
      background: ${theme.colors.primaryColor};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.xlarge} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
    }

    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }

    .image-style-side {
      float: right;
      max-width: 50%;
      margin: ${theme.spacings.medium};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
      width: 100%;
    }
    table td,
    table th {
      padding: ${theme.spacings.small};
    }

    ul,
    ol {
      margin: ${theme.spacings.xlarge};
    }

    @media ${theme.media.lteMedium} {
      font-size: 1.8rem;

      .image-style-side {
        max-width: 100%;
        float: none;
        margin: 0;
      }
    }
  `}
`;
