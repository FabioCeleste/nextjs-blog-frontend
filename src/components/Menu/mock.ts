import { MenuProps } from './index';

export default {
  menuLinks: [
    {
      id: '2',
      link: 'GitHub',
      text: 'muegit',
      newTab: true,
    },
    {
      id: '1',
      link: 'linkedin',
      text: 'meu link',
      newTab: false,
    },
  ],
  attributes: {
    blogName: "Fabio's Blog",
    logo: {
      data: [
        {
          attributes: {
            url: '/assets/images/logo.png',
          },
        },
      ],
    },
  },
} as MenuProps;
