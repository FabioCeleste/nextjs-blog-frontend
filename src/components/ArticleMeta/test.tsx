import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ArticleMeta, ArticleMetaProps } from './index';

import mock from './mock';

const props: ArticleMetaProps = mock;
describe('<ArticleMeta />', () => {
  it('should render author and category links', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(screen.getByRole('link', { name: 'Fabio Celeste' })).toHaveAttribute(
      'href',
      '/author/fabio-celeste',
    );

    expect(screen.getByRole('link', { name: 'JavaScript' })).toHaveAttribute(
      'href',
      '/cetegory/java-script',
    );
  });

  it('should format date', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(screen.getByText('27 de jan. de 2022')).toHaveAttribute(
      'datetime',
      props.createdAt,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleMeta {...props} />);

    expect(container).toMatchSnapshot();
  });
});
