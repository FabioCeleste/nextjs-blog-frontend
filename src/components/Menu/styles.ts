import styled, { css, DefaultTheme } from 'styled-components';

import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  menuVisible: boolean;
  theme: DefaultTheme;
};

const wrapperChanger = (
  menuVisible: MenuBehaviorProps['menuVisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuVisible ? '0' : '-31rem'};
  overflow-y: ${menuVisible ? 'auto' : 'hidden'};

  @media ${theme.media.lteMedium} {
    left: ${menuVisible ? '0' : '-32rem'};
  }
`;

export const Wrapper = styled.div<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    background: ${theme.colors.primaryColor};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;

    ${wrapperChanger(menuVisible, theme)}
  `}
`;

export const Nav = styled.nav`
  ${({ theme }) => css`
    margin: auto;
    width: 100%;
  `}
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: 15px;
    }

    img {
      border: 0.5rem solid ${theme.colors.secondaryColor};
    }
  `}
`;

const buttonChanger = (
  menuVisible: MenuBehaviorProps['menuVisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuVisible ? '26rem' : '1rem'};
  color: ${menuVisible ? theme.colors.secondaryColor : theme.colors.white};
  @media ${theme.media.lteMedium} {
    left: ${menuVisible ? '26rem' : '-0.5rem'};
  }
`;

export const OpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    background: ${theme.colors.primaryColor};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    left: 26rem;
    ${buttonChanger(menuVisible, theme)}
    transition: all 300ms ease-in-out;
  `}
`;
