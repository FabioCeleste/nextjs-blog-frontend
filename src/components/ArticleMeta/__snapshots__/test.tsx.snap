// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleMeta /> should match snapshot 1`] = `
.c0 {
  font-size: 16px;
  color: #aaaaaa;
  font-style: italic;
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <div
    class="c0"
  >
    <p>
      <span>
        Por 
      </span>
      <a
        href="/author/fabio-celeste"
      >
        Fabio Celeste
      </a>
      <span
        class="separator"
      >
         | 
      </span>
      <time
        datetime="2022-01-28T02:53:53.880Z"
      >
        27 de jan. de 2022
      </time>
      <span
        class="separator"
      >
         | 
      </span>
      <span
        class="category"
      >
        <a
          href="/cetegory/java-script"
        >
          JavaScript
        </a>
      </span>
    </p>
  </div>
</div>
`;
